apps/dash-label-properties/app.py

```diff
@@ -0,0 +1,357 @@
+                align="center",
+            ),
+            dbc.Row(
+                dbc.Col(
+                    [
+                        dbc.NavbarToggler(id="navbar-toggler"),
+                        dbc.Collapse(
+                            dbc.Nav(
+                                [dbc.NavItem(button_howto), dbc.NavItem(button_gh)],
+                                className="ml-auto",
+                                navbar=True,
+                            ),
+                            id="navbar-collapse",
+                            navbar=True,
+                        ),
+                    ]
+                ),
+                align="center",
+            ),
+        ],
+        fluid=True,
+    ),
+    color="dark",
+    dark=True,
+)
+
+# Define Cards
+image_card = dbc.Card(
+    [
+        dbc.CardHeader(html.H2("Explore object properties")),
+        dbc.CardBody(
+            dbc.Row(
+                dbc.Col(
+                    dcc.Graph(
+                        id="graph", figure=image_with_contour(img, labels, mode=None),
+                    )
+                )
+            )
+        ),
+    ]
+)
+
+table_card = dbc.Card(
+    [
+        dbc.CardHeader(html.H2("Data Table")),
+        dbc.CardBody(
+            dbc.Row(
+                dbc.Col(
+                    [
+                        dash_table.DataTable(
+                            id="table-line",
+                            columns=columns,
+                            data=table.to_dict("records"),
+                            filter_action="native",
+                            row_deletable=True,
+                            style_table={"overflowY": "scroll"},
+                            fixed_rows={"headers": False, "data": 0},
+                            style_cell={"width": "85px"},
+                        ),
+                        dcc.Store(id="cache", data=labels),
+                        html.Div(id="row", hidden=True, children=None),
+                    ]
+                )
+            )
+        ),
+    ]
+)
+
+app.layout = html.Div(
+    [
+        header,
+        dbc.Container(
+            [dbc.Row([dbc.Col(image_card, md=5), dbc.Col(table_card, md=7)])],
+            fluid=True,
+        ),
+    ]
+)
+
+
+@app.callback(
+    Output("table-line", "style_data_conditional"), [Input("graph", "hoverData")]
+)
+def higlight_row(string):
+    """
+    When hovering hover label, highlight corresponding row in table,
+    using label column.
+    """
+    if not dash.callback_context.triggered:
+        # Nothing has happened yet
+        return []
+    index = string["points"][0]["z"]
+    return [
+        {
+            "if": {"filter_query": "{label} eq %d" % index},
+            "backgroundColor": "#3D9970",
+            "color": "white",
+        }
+    ]
+
+
+@app.callback(
+    [Output("graph", "figure"), Output("cache", "data"), Output("row", "children")],
+    [
+        Input("table-line", "derived_virtual_indices"),
```

Comment:
It looks like it's not possible to select rows (see 
![image](https://user-images.githubusercontent.com/263366/97603495-c9a69780-1a0c-11eb-9331-9aea0f854483.png) 
maybe there is something fishy with the CSS and the selection columns gets cut?
) so this input is never used

Reply:
see my comment above in the definition of the table

Reply:
Also in this callback, I'm not sure anything happens when a table row is deleted

Reply:
![table_interactivity](https://user-images.githubusercontent.com/1302022/97617981-d28b6f00-19f4-11eb-909e-7cf9c235bf09.gif)

- I just tried with the [playground deployment](https://dash-playground.plotly.host/dash-label-properties) and something happens when I select a row, i.e the corresponding element gets a red outline. Selecting a row is however super slow to draw the outline, I had noticed that locally as well. I'm not sure if it has to do with the fact that for every change, we re-build the figure from scratch. Is that what you expected?

- If I delete a table row, the corresponding element is no longer overlayed/highlighted. Is there something else that should happen?

- if I highlight multipe rows (e.g. by shift-click), only the first row gets highlighted. It would be nice to have all regions annotated I think.

Reply:
You are right, the behaviour is correct (selecting one row or deleting rows has the expected behaviour), I just was not patient enough to see the result :-). Performance is indeed an issue here. We can think of several ways of reducing the size: 
- passing the colored overlay as a png string (which should be quite small) in an image trace. You'll have to create a multichannel array for the labels. For this you can use https://datashader.org/api.html#datashader.transfer_functions.shade. However, the down side of passing a colored image instead of a contour plot is that we will not have the index any more in the hover. For this it can be interesting to reproduce what's done in https://scikit-image.org/docs/dev/auto_examples/segmentation/plot_regionprops.html#sphx-glr-auto-examples-segmentation-plot-regionprops-py with filled scatter traces.
- passing the whole `labels` array as `State` and `Output` in the callback is suboptimal, a list of unique labels should be enough.
- finally, it's possible to keep the background image in a store, to have the computed overlay in another store which is the only one used as output in the callback, and to assemble the figure in a clientside callback. I'm hoping that the previous points will optimize enough so that we don't have to resort to clientside callbacks, let's see.